react-app/src/features/package/package-activity/index.tsx

```diff
@@ -126,72 +120,105 @@ const PackageActivity = (props: opensearch.changelog.Document) => {
+          {packageActivity.timestamp
+            ? format(new Date(packageActivity.timestamp), "eee, MMM d, yyyy hh:mm:ss a")
             : "Unknown"}
         </p>
       </AccordionTrigger>
       <AccordionContent className="p-4">
-        <Submission {...props} />
+        <Submission packageActivity={packageActivity} />
       </AccordionContent>
     </AccordionItem>
   );
 };
 
+type DownloadAllButtonProps = {
+  packageId: string;
+  submissionChangelog: ItemResult[];
+};
+
+const DownloadAllButton = ({ packageId, submissionChangelog }: DownloadAllButtonProps) => {
+  const { onZip, loading } = useAttachmentService({ packageId });
+
+  if (submissionChangelog.length === 0) {
+    return null;
+  }
+
+  const onDownloadAll = () => {
+    const attachmentsAggregate = submissionChangelog.reduce<Attachments>((acc, changelogItem) => {
+      if (!changelogItem._source.attachments) {
+        return acc;
+      }
+
+      return acc.concat(changelogItem._source.attachments);
+    }, []);
+
+    if (attachmentsAggregate.length === 0) {
+      return;
+    }
+
+    onZip(attachmentsAggregate);
+  };
+
+  return (
+    <Table.Button loading={loading} onClick={onDownloadAll} variant="outline">
+      Download all documents
+    </Table.Button>
+  );
+};
+
 export const PackageActivities = () => {
-  const {
-    data: packageActivity,
-    loading,
-    onDownloadAll,
-    accordianDefault,
-  } = usePackageActivities();
-
-  const activitiesWithoutAdminChange = packageActivity.filter(
+  const { id: packageId } = useParams<{ id: string }>();
+  const { data: submission } = useGetItem(packageId);
+
+  if (submission === undefined) {
+    return null;
+  }
+
+  const submissionChangelogWithoutAdminChanges = submission._source.changelog.filter(
     (activity) =>
-      activity._source.isAdminChange === undefined || activity._source.isAdminChange === false,
+      // isAdminChange can be `undefined` or `boolean`
+      !activity._source.isAdminChange,
   );
 
+  const keyAndDefault = submissionChangelogWithoutAdminChanges[0]?._source?.id;
+
   return (
     <DetailsSection
       id="package_activity"
       title={
         <div className="flex justify-between">
-          Package Activity ({activitiesWithoutAdminChange.length})
-          {activitiesWithoutAdminChange.length && (
-            <Table.Button loading={loading} onClick={onDownloadAll} variant="outline">
-              Download all documents
-            </Table.Button>
-          )}
+          Package Activity ({submissionChangelogWithoutAdminChanges.length})
+          <DownloadAllButton
+            submissionChangelog={submissionChangelogWithoutAdminChanges}
+            packageId={packageId}
+          />
         </div>
       }
     >
-      {activitiesWithoutAdminChange.length === 0 && (
+      {submissionChangelogWithoutAdminChanges.length > 0 ? (
+        <Accordion
+          // `key` to re-render the `defaultValue` whenever `accordionDefault` changes
```

Comment:
is accordionDefault/accordianDefault still valid? cant find it

Reply:
Resolving with https://github.com/Enterprise-CMCS/macpro-mako/pull/859/commits/8976a4c9d699c7c90b39cfad8d51c3cf4108e4ce